common/ast/ast.go

```diff
@@ -343,21 +333,26 @@ func (s *SourceInfo) GetStartLocation(id int64) common.Location {
 // be identical to the start location for the expression.
 func (s *SourceInfo) GetStopLocation(id int64) common.Location {
 	if o, found := s.GetOffsetRange(id); found {
-		line := 1
-		col := int(o.Stop)
-		for _, lineOffset := range s.LineOffsets() {
-			if lineOffset < o.Stop {
-				line++
-				col = int(o.Stop - lineOffset)
-			} else {
-				break
-			}
-		}
-		return common.NewLocation(line, col)
+		return s.GetLocationByOffset(o.Stop)
 	}
 	return common.NoLocation
 }
 
+// GetLocationByOffset returns the line and column information for a given character offset.
+func (s *SourceInfo) GetLocationByOffset(offset int32) common.Location {
+	line := 1
+	col := int(offset)
+	for _, lineOffset := range s.LineOffsets() {
+		if lineOffset <= offset {
+			line++
+			col = int(offset - lineOffset)
+		} else {
+			break
+		}
```

Comment:
super nit: structuring this with a guard clause would make this read better imo. Up to you if you want to make the change:

```suggestion
        if lineOffset > offset {
            break
        }
        line++
        col = int(offset - lineOffset)
```